bundle/src/main/java/com/meltmedia/dropwizard/etcd/cluster/ProcessorStateTracker.java

```diff
@@ -0,0 +1,138 @@
+package com.meltmedia.dropwizard.etcd.cluster;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.SortedSet;
+import java.util.function.BiFunction;
+
+import com.google.common.collect.Ordering;
+import com.google.common.collect.Sets;
+import com.meltmedia.dropwizard.etcd.json.EtcdEvent;
+import com.meltmedia.dropwizard.etcd.json.EtcdJson.MappedEtcdDirectory;
+import com.meltmedia.dropwizard.etcd.json.WatchService.Watch;
+
+public class ProcessorStateTracker {
+  public static class Builder {
+    private MappedEtcdDirectory<ProcessorNode> directory;
+    private ClusterNode thisNode;
+
+    public Builder withDirectory(MappedEtcdDirectory<ProcessorNode> directory) {
+      this.directory = directory;
+      return this;
+    }
+
+    public Builder withThisNode(ClusterNode thisNode) {
+      this.thisNode = thisNode;
+      return this;
+    }
+
+    public ProcessorStateTracker build() {
+      return new ProcessorStateTracker(directory, thisNode);
+    }
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  private MappedEtcdDirectory<ProcessorNode> directory;
+  private Watch watch;
+  private volatile ProcessorState state;
+  private ClusterNode thisNode;
+  
+  private ProcessorStateTracker(MappedEtcdDirectory<ProcessorNode> directory, ClusterNode thisNode) {
+    this.directory = directory;
+    this.state = ProcessorState.empty();
+    this.thisNode = thisNode;
+  }
+  
+  public void start() {
+    watch = directory.registerWatch(this::handle);
+  }
```

Comment:
This and `stop()` should take runnables, since proper starting and stopping requires action before and after other processor components are set up.